### DISCUSSION TITLE
How to cross compile oqsprovider and its dependencies liboqs/openssl for arrch64

### DISCUSSION THREAD

**Discussion by ranjankumar23** [open-quantum-safe/oqs-provider#489] [General]
I am trying to use scripts/fullbuild.sh to cross compile oqsprovider and its dependencies liboqs/openssl for arrch64.

But i don't see option to provide toolchain details as it is there for liboqs/openssl-1.1.1 using '-DCMAKE_TOOLCHAIN_FILE=../.CMake/toolchain_arm64.cmake'.

Need your help.

**Comment by cothan** [comment]
Just one quick note, `oqs-provider` need at least OpenSSL 3.2 to work. Can you check again with newer version of OpenSSL?

**Reply by baentsch** [reply]
Err, `oqsprovider` is designed to work with all `openssl` v3.xx versions; just TLS-sigs don't work below 3.2. Please let me know if you have proof to the opposite (see also #488), @cothan

**Reply by cothan** [reply]
Thanks for the correction @baentsch .

**Comment by baentsch** [comment]
> i don't see option to provide toolchain details

@ranjankumar23 there are no such options for those files. It would be a nice addition, though, if you'd like to contribute. Until then, you may want to take a look at the [CI scripts for `aarch64`](https://github.com/open-quantum-safe/oqs-provider/blob/a0f475daba6dcbe3088ff8943810d15cecd2c971/.github/workflows/linux.yml#L165-L234)
